presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -346,13 +355,35 @@ private void addRolesRecords()
         }
     }
 
+    private void addRoleAuthorizationDescriptorRecords()
+    {
+        try {
+            accessControl.checkCanShowRoles(session.toSecurityContext(), catalogName);
```

Comment:
I think we need dedicated access control method for that.

Reply:
I was thinking about that. Personally I thought there's access to roles or not.

One could argue that access to ROLE_AUTHORIZATION_DESCRIPTORS is a superset access right (I can get the list of roles from that table as well.) 
Separate access rights seems a bit overkill - happy to put it in if you feel strongly, though.

Reply:
Presto SPI do not assume much about authorization system implementation. Having separate method allows it to decide how it should behave. I would prefer to have separate method.

Reply:
OK... Added that.